core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -123,12 +130,34 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf:SparkConf = null
+
   override def open(): BlockObjectWriter = {
     if (hasBeenClosed) {
       throw new IllegalStateException("Writer already closed. Cannot be reopened.")
     }
     fos = new FileOutputStream(file, true)
-    ts = new TimeTrackingOutputStream(fos)
+    var isEncryptedShuffle: Boolean = if (sparkConf != null) {
```

Comment:
@vanzin : i will change "var" to "val".  Use if ... else here is to pass [ the unit test ("Reopening a closed block writer") of BlockObjectWriterSuite ](https://github.com/apache/spark/blob/dcd1e42d6b6ac08d2c0736bf61a15f515a1f222b/core/src/test/scala/org/apache/spark/storage/BlockObjectWriterSuite.scala#L80). SparkConf is null in that unit test.